### ISSUE TITLE
version 1.7.1 crashes on iphone

### ISSUE THREAD

**Issue by shiminghuang** [MiniKeePass/MiniKeePass#554]
when open my database which was copied from iCloud, minikeepass flash exits. That database was created in windows PC and saved to iCloud for iOS use. My windows keepass version is 2.36.
I got this problem after updated to the new version of 1.7.1.
It can open the other database which was created inside iOS miniKeepass.
please help to solve this.
thank you.

**Comment by shiminghuang** [comment]
iOS is on iphone 7 plus, version 10. 3.3

**Comment by nosacs** [comment]
Using iPhone SE w/ ios11; after upgrade to minikeepass 1.7.1 can no longer open database.  Open results in minimized minikeepass at the database selection.  Database was initially created via KeePass on a Windows platform, then transported to KeePass running on OSx.  Database has been working perfectly between OSx, export/import to minikeepass via Dropbox until this last upgrade to 1.7.1.  This is a real pain since I am no unable to access my passwords.

**Comment by ziggy10** [comment]
Same situation: ios11.0.1/Windows 10.
Keepass 2.36 with KeepassLibC 1.33 and 1.x File Support on Windows
minikeepass 1.7.1 on ios.
Database created and maintained on Windows as a master. 
On Windows everthing is fine.
Sharing with ios devices per owncloud. 
On ios when searching from database "root" minikeepass crashes. 
Also when trying to open entries in some groups. 
In other groups searching within the group and opening entries within this group is ok.

**Comment by dabadur** [comment]
Same here: iPhone 6s, ios 11.0.2, minikeepass 1.7.1
I can no longer open dataspace. During selecting keyfile the app crashes.
Big THX for fixing that,
dabadur

**Comment by gabox7** [comment]
Ver.1,7.2 maybe works smothly ;) Ill test during this day. thanks in advance for fast fixed..

**Comment by noraisk** [comment]
Thanks, in 1.7.2 it works again.

**Comment by nosacs** [comment]
Thanks!  1.7.2 is working.  Great response!

**Comment by hcarrega** [comment]
It’s working now

**Comment by ziggy10** [comment]
everthing is perfect now. Thanks a lot.

**Comment by dabadur** [comment]
Yeah, it`s working now (1.7.2). But: if i have to scroll through the fake-(key-)filelist, minikeepass crashes again. So i reduced the files … everything is fine then. :o

**Comment by adamtait** [comment]
Crash on selecting keyfile.
version: 1.7.1 & 1.7.2 (still crashing in latest update)
device: iPhone 5, iOS version 10.3.3

I have a collection of databases & keys on my Minikeepass install. I can select any database file, but when I go to select a keyfile then Minikeepass crashes.
I've done some of my own testing and found that the crashes aren't entirely consistent. If I select enough different keyfile, I can always get Minikeepass to crash. However, it won't crash consistently on the same keyfile. I may open up the app and select a keyfile without incident, but it will crash if I try the same sequence again.

**Comment by tssmith7** [comment]
@adamtait There is a bug that causes MKP to crash if you scroll the currently selected key file off the screen and select another key file.  (line `57` of `KeyFileViewController.swift`)
